### ISSUE TITLE
Rewrite using gnome technologies instead of electron? 🤔 

### ISSUE THREAD

**Issue by edipox** [edipox/wunderlistux#18]
Rewrite using gnome (gjs, vala, etc) techs instead of electron?

**Comment by handicraftsman** [comment]
Would be great.  P. S. You can try using Glade.

**Comment by edipox** [comment]
Thank you for your suggestion @handicraftsman

**Comment by handicraftsman** [comment]
@edipox, by using glade you can make GUI using XML and write app using almost any PL (with available GTK bindings). Yes, ruby too.

**Comment by edipox** [comment]
@handicraftsman That's very cool, I didn't know it. Thank you! I'll definitely take a look at Glade and PLs with GTK bindings

**Comment by DroWnThePoor** [comment]
Is someone actively developing this? Because that would be awesome.

---- Edipo Vinicius da Silva wrote ----

**Comment by edipox** [comment]
@DroWnThePoor Not yet. There are a lot of options. I'll find some time and do some research, but for now there is still no code for this.

**Comment by edipox** [comment]
To everyone reading this: Any idea, code and suggestions about this are more than welcome

**Comment by edipox** [comment]
Hi guys, look at this 
![vala](https://cloud.githubusercontent.com/assets/1580541/21660091/b0adc33a-d2ac-11e6-9ccd-c4bf6fe64dbc.png)

If you want to check it out you can download the code from the [vala branch](https://github.com/edipox/wunderlistux/tree/vala_webkit) and run:
`mkdir build ; cd build && cmake .. -DCMAKE_INSTALL_PREFIX=/usr && make && sudo make install`

**Comment by handicraftsman** [comment]
Nice. BTW icons should be a bit smaller.

**Comment by handicraftsman** [comment]
Also window is too large.

**Comment by handicraftsman** [comment]
Also try that to get rid of builtin webkit and use less resources: [link](https://developer.wunderlist.com/)

**Comment by edipox** [comment]
Hi @alphaphee ! We are working on a Wunderlistux refactor with gnome/elementary technologies and it would be great if we could receive some icon love :)

**Comment by alphaphee** [comment]
I am pretty happy to help with this. I was actually looking for an elementary wunderlist app (even built around electron).

**Comment by alphaphee** [comment]
![48px](https://cloud.githubusercontent.com/assets/4603691/26210832/8e3e94bc-3bf9-11e7-9733-dcd363ee3be1.png)

So would you prefer that the icon follows the original wunderlist icon style?

**Comment by edipox** [comment]
@alphaphee it's ok if it follows the original Wunderlist style, but I think it'd be nice to have something a bit different. Anyway, it's pretty much up to you. I've seen the icons you posted on g+ and your work is awesome

**Comment by dignityinside** [comment]
+1

+ You win performance an reduce the size of the Linux app

- You can not create crossplattform app without electron - you lose people who use macOS or WIndows (but i think this people use the official app).

**Comment by edipox** [comment]
Yes @dignityinside, a native app is the next step for Wunderlistux for sure. 
We will add @alphaphee icon and finish the work on the Vala branch which is based on Webkit or maybe we will go ahead and replace Webkit for an even 'more native' app, using hooks to communicate with the wunderlist API. 
After that, it will be published on the elementary AppCenter.
And the final step I think would be to implement syncing, for offline usage.

**Comment by mateossh** [comment]
Is it worth maintaining something that will be shut down soon?

**Comment by edipox** [comment]
@mateossh Probably it isn't. But I'm not sure if I understood what you are referring to. Would you explain?

**Comment by tuaminx** [comment]
I think @mateossh mentioned MS who purchased Wunderlist. And they provide Microsoft To-do app.

**Comment by edipox** [comment]
Yes @tuaminx , I saw that after @mateossh comment. Still, I hope this migration from Wunderlist to MS To-do will take a while. In any case, they will probably provide an API as Wunderlist does and, theoretically, it would be possible for us to move to the MS to-do as well. But to be honest I'm not sure if I'm into that. I do like Wunderlist but MS Todo?.. I'm not sure.

**Comment by handicraftsman** [comment]
@dignityinside 

> You can not create crossplattform app without electron - you lose people who use macOS or WIndows (but i think this people use the official app).

GTK works on all Linux, Windows and OSX. You need to include GTK, GLib, and other library binaries to port it to other systems though.

**Comment by ryukinix** [comment]
This is pretty nice. I like it and prefer that instead the big electron app.

**Comment by ryukinix** [comment]
> GTK works on all Linux, Windows and OSX. You need to include GTK, GLib, and other library binaries to port it to other systems though.

Yes. GTK is portable, but is GLib-portable and we have GLib support for a lot of systems.

**Comment by handicraftsman** [comment]
Tried it. Looks nice:
![2017-11-06 09-28-57](https://user-images.githubusercontent.com/6382804/32429818-105afb2c-c2d5-11e7-93c5-1cbb9acc0d01.png)

**Comment by ryukinix** [comment]
I've created a AUR (Arch User Repository) package for the `vala_webkit` branch here: https://aur.archlinux.org/packages/wunderlistux-gtk/

If someone use Arch Linux here as I use (or any Arch-based distro), please would be nice a feedback if this works for you (or not, and providing a comment here on the aur package page why not and details)

**Comment by ryukinix** [comment]
I have, otherwise, a few bugs on this version `vala_webkit`:

+ weirdly I got old timestamps for my To-Do tasks (due-date wrong)
+ some process of webkit2gtk are created on the session of `wunderlistux` but not killed when the window is closed,    this leaves some zombie process on the end which I need to kill manually.

I should fill properly issues about that or you don't have more interesting in support this version, @edipox?

**Comment by sandangel** [comment]
Hi everyone, any progress on vala_webkit version bugs?

**Comment by edipox** [comment]
Hi @sandangel, unfortunately, we have received no additional contributions nor updates 😞
